src/borrow_tracker/tree_borrows/wildcard.rs

```diff
@@ -0,0 +1,592 @@
+            let mut old_child_access = src_old_local_access;
+            // and after this update.
+            let mut new_child_access = src_new_local_access;
+            while let Some(parent_id) = nodes.get(child).unwrap().parent {
+                let parent_node = nodes.get(parent_id).unwrap();
+                let mut entry = wildcard_accesses.entry(parent_id);
+                let parent_state = entry.or_insert(Default::default());
+
+                let old_parent_state = parent_state.clone();
+                use WildcardAccessLevel::*;
+                // Updating this node's tracking state for its children.
+                if is_upgrade {
+                    if new_child_access == Write {
+                        // None -> Write
+                        // Read -> Write
+                        parent_state.child_writes += 1;
+                    }
+                    if old_child_access == None {
+                        // None -> Read
+                        // None -> Write
+                        parent_state.child_reads += 1;
+                    }
+                } else {
+                    if old_child_access == Write {
+                        // Write -> None
+                        // Write -> Read
+                        parent_state.child_writes -= 1;
+                    }
+                    if new_child_access == None {
+                        // Read  -> None
+                        // Write -> None
+                        parent_state.child_reads -= 1;
+                    }
+                }
+
+                let old_parent_local_access = old_parent_state.max_local_access();
+                let new_parent_local_access = parent_state.max_local_access();
+
+                {
+                    // We need to update the `max_foreign_access` of `child`'s siblings.
+                    // For this we can reuse the `push_relevant_children` function.
+                    //
+                    // We pass it just the siblings without child itself. Since `child`'s
+                    // `max_local_access()` is foreign to all of its siblings we can pass
+                    // it as part of the foreign access.
+
+                    // it doesnt matter wether we read these from old or new
+                    let constant_factors =
+                        max(parent_state.exposed_as, parent_state.max_foreign_access);
+
+                    // `state` contains the correct child_writes/reads counts for just the
+                    // siblings excluding `child`.
+                    let state = if is_upgrade { &old_parent_state } else { parent_state };
+
+                    Self::push_relevant_children(
+                        &mut stack,
+                        max(constant_factors, new_child_access),
+                        max(constant_factors, old_child_access),
+                        state.child_reads,
+                        state.child_writes,
```

Comment:
It looks like this is always `min(old_parent_state.child_reads, parent_state.child_reads)`? Is that correct? Not that I understand why that's the right thing to pass here, still needs a more detailed comment.